### PR TITLE
feat(tupaiaWeb): RN-1305: Map overlay exports

### PR DESCRIPTION
**Pull request by alexd-bes**
### Changes:
- Added `ExportMapOverlayRoute` to `tupaia-web-server`
- Added export button to map overlay selector (desktop only)
- Added export route

---

### Screenshots:
Screenshot 2024-06-04 at 4 20 54 PM
Screenshot 2024-06-04 at 4 21 08 PM

### CODE REVIEW

File: packages/ui-components/src/components/PDFExportComponent.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const A4_PAGE_HEIGHT_MM = 297;
  * use absolute length units (like cm and pt) and remove these constants at some point.
  */
 export const A4_PAGE_WIDTH_PX = 1191; // at 144ppi
-export const A4_PAGE_HEIGHT_PX = 1684; // at 144ppi
+export const A4_PAGE_HEIGHT_PX = 1683; // at 144ppi
```
Comment (alexd-bes) [inline packages/ui-components/src/components/PDFExportComponent.tsx:17]: This was rounded up from a decimal number before, and this caused landscape pages to overflow vertically

File: packages/ui-map-components/src/components/Legend/MarkerLegend.tsx [hunk 2]
```diff
@@ -27,16 +27,22 @@ import {
...
   justify-content: center;
   flex-wrap: wrap;
 
-  ${p => p.theme.breakpoints.down('sm')} {
-    flex-direction: column;
+  ${({ $isExport, theme }) =>
+    !$isExport &&
+    `
+  ${theme.breakpoints.down('sm')} {
+    flex-direction:  column;
```
Comment (jaskfla) [inline packages/ui-map-components/src/components/Legend/MarkerLegend.tsx:42]: This one might need to go through Prettier again

```suggestion
    css`
  ${theme.breakpoints.down('sm')} {
    flex-direction: column;
```

**Review by jaskfla — APPROVED** [review]
Impressive stuff, screenshot’s looking great! So much for a 13-pointer 😆

Do you know if we anticipate people printing these exports? If so, the “Latest overlay data” text in grey might be a little too light to come out well in print

Couple inconsequential details, and just one file that I’ll let you decide if you wanted Rohan’s review on

File: packages/tupaia-web/src/api/mutations/useExportMapOverlay.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    ({
+      projectCode,
+      entityCode,
+      mapOverlayCode,
+      zoom,
+      center,
+      hiddenValues,
+      tileset,
+    }: ExportDashboardBody) => {
+      const baseUrl = `${window.location.protocol}//${window.location.host}`;
```
[the remark this answers is not in the document]
Comment (alexd-bes) [reply packages/tupaia-web/src/api/mutations/useExportMapOverlay.ts:35]: Yep, it's the `http` or `https` part of the location

File: packages/types/src/types/requests/tupaia-web-server/ExportMapOverlayRequest.ts
[the remark this answers is not in the document]
Comment (alexd-bes) [reply packages/types/src/types/requests/tupaia-web-server/ExportMapOverlayRequest.ts:1]: I'm going to keep it as a `.ts` to match the pattern elsewhere in this folder. It also generates a `.d.ts` file on build anyway

**Pull request merged**